client/src/components/ProductDetails/index.jsx

```diff
@@ -8,56 +10,33 @@ import ReviewStars from '../common/ReviewStars';
 import SocialShare from './SocialShare';
 import ProductPrice from './ProductPrice';
 
+
 const ProductDetails = ({ productId }) => {
-  const [productInfo, setProductInfo] = useState();
-  const [productStyles, setProductStyles] = useState();
+
   const [selectedStyle, setSelectedStyle] = useState();
-  const [numReviews, setNumReviews] = useState();
-  const [reviewRating, setReviewRating] = useState();
+
+  const productInfo = useSelector(
+    (state) => state.productInfo.data?.[productId]
+  );
+  const productStyles = useSelector(
+    (state) => state.productStyles.data?.[productId]
+  );
+  const reviewMeta = useSelector((state) => state.reviewMeta.data?.[productId]); 
+  const numReviews = reviewMeta?.ratingsSummary?.count; 
+  const reviewRating = reviewMeta?.ratingsSummary?.average;
+ 
```

Comment:
I see that in Redux states are replaced by useSelectors which i assume gets the state from the central store